views-core/src/main/java/io/micronaut/views/ViewsFilter.java

```diff
@@ -75,107 +71,63 @@ public int getOrder() {
                                                             ServerFilterChain chain) {
 
         return Flux.from(chain.proceed(request))
-            .switchMap(response -> {
-                Optional<AnnotationMetadata> routeMatch = response.getAttribute(HttpAttributes.ROUTE_MATCH,
-                        AnnotationMetadata.class);
-                if (routeMatch.isPresent()) {
-                    AnnotationMetadata route = routeMatch.get();
+                .switchMap(response -> {
```

Comment:
I did a medium size refactor of the ViewFilter, putting most exits at the top of the method and simplifying the logic (by removing as much of it as possible)